include/oneapi/internal/execution_context_sycl.h

```diff
@@ -113,20 +148,42 @@ class OpenClKernelFactory : public Base, public ClKernelFactoryIface
         }
         else
         {
-            auto kernelRef = OpenClKernelRef(_currentProgramRef->get(), kernelName, &localStatus);
-            if (!localStatus.ok())
+            if (!_deviceQueue.get_device().template get_info<sycl::info::device::opencl_c_version>().empty())
```

Comment:
I suggest to reorganize it this way:
```cpp
KernelPtr kernel;
if (...) {
    ...
    kernel.reset(new OpenClKernelNative(_executionTarget, *_currentProgramRef, kernelRef));
}
else {
    ...
    kernel.reset(new OpenClKernelLevel0(_executionTarget, *_currentProgramRef, kernelRef));
}

kernelHashTable.add(key, kernel, localStatus);
if (!localStatus.ok()) {
    services::internal::tryAssignStatus(status, localStatus);
    return KernelPtr();
}
return kernel;
```

Reply:
Minor comment: I suggest to not use "negative if"
```
if (!const) { A } else { B } // not very good
if (cond) { B } else { A } // good
```

Reply:
It's difficult to understand the condition like `_deviceQueue.get_device().template get_info<sycl::info::device::opencl_c_version>().empty()`, it's worth to define a variable or method, like this 

```cpp
const bool isOpenCLBackendAvailable = _deviceQueue.get_device().template get_info<sycl::info::device::opencl_c_version>().empty()
```

Reply:
fixed